### ISSUE TITLE
Column collapsing failing for Renaud Leo

### ISSUE THREAD

**Issue by dancunningham** [projectbuendia/client#233]
It's showing over 100 columns for him, making chart  loading painfully slow and the chart unusable.

Maybe something to do with orders?

**Comment by capnfabs** [comment]
@dancunningham When there's an order spanning multiple days, we intentionally don't collapse the columns, even if there are no observations. We can change that but I'm assuming that it was done for a reason to begin with. It's important that active orders that weren't given are indicated as such, right?

**Comment by capnfabs** [comment]
Previous discussion: https://github.com/projectbuendia/client/pull/72, especially https://github.com/projectbuendia/client/pull/72#issuecomment-152348199

Agreed that the performance is abysmal though. And the one-minute refresh is an absolute killer, because the auto-scroll thing can't work properly because the HTML page is too large.

We can make it so old orders with no observations are collapsed, I personally don't think it's worth it though - it's a pretty unusual case.

**Comment by dancunningham** [comment]
This is also a duplicate of https://github.com/projectbuendia/client/issues/208

**Comment by dancunningham** [comment]
@capnfabs you're right that it's important to see that orders were prescribed but not given, so we shouldn't simply collapse columns where an order was ordered but not recorded as given.

I have deleted the order causing this for this patient, and now his chart works fine again.

This case should not come up during the pilot and generally shouldn't come up as patients are only in ITFC for a max of about 2 weeks, so nothing should be ordered over that long a timescale.

We can deprioritise this for post-pilot, though it does need to be fixed and for other conditions like HIV / TB where treatments do go on over months/years, this definitely needs addressed (along with more flexible ways to prescribe orders e.g. with future start dates)

**Comment by llvasconcellos** [comment]
@capnfabs I fixed the missing code referred on Ping's comment.

**Comment by capnfabs** [comment]
@llvasconcellos yeah I know :) Sorry, I should've been more clear about that. I was trying to say that it was an intentional decision that was agreed upon by Dan, Leonardo, and Ping.

**Comment by llvasconcellos** [comment]
@capnfabs :+1:
